File: backend/SkillProof/SkillProof.Entities/Dtos/Questions/QuestionListFilterDto.cs
using SkillProof.Entities.Enums;

namespace SkillProof.Entities.Dtos.Questions
{
    public class QuestionListFilterDto
    {
        public QuestionType? Type { get; set; }
        public DifficultyLevel? Difficulty { get; set; }
        public string? Language { get; set; }
        public bool? IsActive { get; set; }
    }
}

File: web/src/app/Models/Dtos/Question/question-list-filter-dto.ts
import { DifficultyLevel } from '../../Enums/DifficultyLevel';
import { QuestionType } from '../../Enums/QuestionType';

export class QuestionListFilterDto {
  type?: QuestionType;
  difficulty?: DifficultyLevel;
  language?: string;
  isActive?: boolean;
}
